File: tests/mixedvector.c
#include "head.h"

extern void whileIf(long, float *, float *);
extern void ifWhile(long, float *, float *);
extern void moreNested(long, float *, float *, float *);
extern void stress1(long, float *, float *);

int main(void) {
  int num = 8;
  int i = 0;
  float *a = createvect(num),
        *b = createvect(num),
        *c = createvect(num);
  
  // while with an if loop inside it
  setvect(num, a, 0);
  setvect(num, b, 0);
  for (i = 0; i < num; i ++){
    a[i] = i;
    b[i] = i*i;
  }
  whileIf(num, a, b);
  printf("0: a = ");
  printvect(num, a);
  printf("\n");
  printf("   b = ");
  printvect(num, b);
  printf("\n");
 
  // such that we go into the while loop
  setvect(num, a, 0);
  setvect(num, b, 0);
  for (i = 0; i < num; i ++){
    a[i] = i-10;
    b[i] = i-5;
  }
  whileIf(num, a, b);
  printf("1: a = ");
  printvect(num, a);
  printf("\n");

  // if with a while loop inside it
  setvect(num, a, 0);
  setvect(num, b, 0);
  for (i = 0; i < num; i ++){
    a[i] = i;
    b[i] = i*i;
  }
  ifWhile(num, a, b);
  printf("2: a = ");
  printvect(num, a);
  printf("\n");
  printf("   b = ");
  printvect(num, b);
  printf("\n");

  // if with a while loop inside it
  setvect(num, a, 0);
  setvect(num, b, 0);
  for (i = 0; i < num; i ++){
    a[i] = i-10;
    b[i] = i-5;
  }
  ifWhile(num, a, b);
  printf("3: a = ");
  printvect(num, a);
  printf("\n");
  printf("   b = ");
  printvect(num, b);
  printf("\n");
  
  // more nested brackets 
  setvect(num, a, 0);
  setvect(num, b, 0);
  setvect(num, c, 0);
  for (i = 0; i < num; i ++){
    a[i] = i;
    b[i] = i*i;
    c[i] = i*2;
  }
  moreNested(num, a, b, c);
  printf("4: a = ");
  printvect(num, a);
  printf("\n");
  printf("   b = ");
  printvect(num, b);
  printf("\n");
  printf("   c = ");
  printvect(num, c);
  printf("\n");

  // stress test 1 into the first if
  num = 32;
  setvect(num, a, 0);
  setvect(num, b, 0);
  for (i = 0; i < num; i ++){
    a[i] = i;
    b[i] = i*i;
  }
  stress1(num, a, b);
  printf("5: a = ");
  printvect(num, a);
  printf("\n");
  printf("   b = ");
  printvect(num, b);
  printf("\n");

  return 0;
}

File: tests/mixedvector.py
a = []
b = []
c = []
for i in xrange(0,32):
    a.append(float(i))
    b.append(float(i*i))
    c.append(float(i*2))

def whileIf(a,b):
    while sum(a) < 10:
        print "hi"
        if sum(b) < 5:
            for i in xrange(len(b)):
                b[i] = b[i] * b[i]
        else:
            for i in xrange(len(b)):
                b[i] = b[i] - 1
        for i in xrange(len(a)):
            a[i] = a[i] + 1
        
def ifWhile(a,b):
    if sum(b) < 4:
        while sum(a) < 5:
            for i in xrange(len(a)):
                a[i] = a[i] + 1
                b[i] = b[i]*3
            for i in xrange(len(b)):
                b[i] = min(b[i],20)
        for i in xrange(len(a)):
            a[i] = 2
    else:
        for i in xrange(len(a)):
            a[i] = 0
            b[i] = 0

def nested(a,b,c):
    d = []
    for i in xrange(len(a)):
        d.append(min (min(a[i],b[i]),c[i]) -1)
        b[i] = 5*(1+min(a[i],b[i])/2)+2
        a[i] = d[i] + a[i] - c[i]
        c[i] = 2./4.
        
def stress(a,b):
    x = []
    for i in xrange(len(a)):
        x.append(3*(min(a[i],b[i]) +1))
    if sum(x) < 100:
        for i in xrange(len(a)):
            a[i] = 10
        while sum(a) < 10:
            for i in xrange(len(a)):
                b[i]=b[i]+1
                a[i]=a[i]-1
    else:
        for i in xrange(len(a)):
            a[i]=x[i]
            b[i]=x[i]
    for i in xrange(len(a)):
        a[i] = b[i]*a[i]*b[i]

stress(a,b)
print a
print b
